### PR TITLE
tests: subsys: Check zephyr os tracing functions.

### PR DESCRIPTION
**Pull request by Zhaoningx**
Add some testcases for subsys/trace and Use
self-define backends to get tracing data
and check the output data.

Signed-off-by: Ningx Zhao <ningx.zhao@intel.com>

### CODE REVIEW

File: tests/subsys/tracing/tracing_sys/src/main.c [hunk 1]
```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
```
Comment (enjiamai) [inline tests/subsys/tracing/tracing_sys/src/main.c:2]: Please change the year to 2021. So does  "tests/subsys/tracing/tracing_sys/src/tracing_string_format_test.c: line 2"

File: tests/subsys/tracing/tracing_sys/src/main.c [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+ * Expected Test Result:
+ * - Tracing data include thread,idle thread info
+ * based on the provided trace data.
+ *
+ * Pass/Fail Criteria:
+ * - Successful if check points in test procedure are all passed, otherwise failure.
+ *
+ * Assumptions and Constraints:
+ * - N/A
+ */
```
Comment (enjiamai) [inline tests/subsys/tracing/tracing_sys/src/main.c:135]: Excuse me, could you please remove this kind of comment at this moment? Just keep good "brief, detail, ingroup, see, etc" in place.

File: tests/subsys/tracing/tracing_sys/src/main.c [hunk 3]
```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <ztest.h>
+#if defined(CONFIG_TRACING_BACKEND_UART) || defined(CONFIG_TRACING_BACKEND_USB)
+#include "../../../../subsys/tracing/include/tracing_backend.h"
```
Comment (enjiamai) [inline tests/subsys/tracing/tracing_sys/src/main.c:8]: Can we make it shorter, just like "include/tracing_backend.h" or something like this?

**Review by enjiamai — CHANGES_REQUESTED** [review]

File: tests/subsys/tracing/tracing_sys/src/main.c (hunk 1 above)
Comment (Zhaoningx) [reply tests/subsys/tracing/tracing_sys/src/main.c]: Done, thanks

File: tests/subsys/tracing/tracing_sys/src/main.c (hunk 3 above)
Comment (Zhaoningx) [reply tests/subsys/tracing/tracing_sys/src/main.c:8]: It doesn't work if I check it, maybe need some time to take a look at this, thanks.

**Review by enjiamai — APPROVED** [review]

File: tests/subsys/tracing/tracing_sys/src/main.c [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+ * should be false.
+ *
+ * @ingroup driver_sensor_subsys_tests
+ */
+void test_tracing_function(void)
+{
+#ifdef CONFIG_TRACING_CTF
+	ztest_test_skip();
+#endif
+	generate_more_tracing_data();
```
Comment (LeiW000) [inline tests/subsys/tracing/tracing_sys/src/main.c:111]: Here, we need to make sure that "thread" already exits.

**Review by LeiW000 — CHANGES_REQUESTED** [review]

File: tests/subsys/tracing/tracing_sys/src/main.c (hunk 4 above)
Comment (LeiW000) [reply tests/subsys/tracing/tracing_sys/src/main.c:124]: Have you addressed this comment with the latest commit?

Comment (Zhaoningx) [reply tests/subsys/tracing/tracing_sys/src/main.c:124]: Hi @LeiW000 , I added a API k_thread_join() in the funciton generate_more_tracing_data, and the sub-thread has already  exited before getting here. I don't know if I fixed this comment.

**Comment by Zhaoningx** [comment]
Hi @nashif , would you like to take a look at this PR, thanks.

**Review by LeiW000 — APPROVED** [review]

**Comment by Zhaoningx** [comment]
Hi @nashif @andyross , I think I have addressed all comments, and the testcase is used to test tracing function by using the self-designed tracing backend on qemu_x86 platform, and you know, zephyr branch will be released on date 5.7, so I want to got merged before this deadline, so would you like to take a look at this PR, and to comment it if there are anything wrong, thanks very much.

**Comment by nashif** [comment]
marking as DNM, we expect some major changes to tracing subsystems coming from #33603, will revisit and integrate with new changes once they land.

**Pull request closed without merging**